### PR TITLE
Update the container registry examples for `steps.uses`

### PR DESCRIPTION
**Pull request by varunsridharan**
### Why:
["Example using a Docker public registry action"](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-a-docker-public-registry-action) needs to explain which container registry is being demonstrated.

### What's being changed:

- ["Example using a Docker public registry action"](https://docs-1121--patch-1.herokuapp.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-a-docker-public-registry-action) - Updated example to clarify which container registry service is being used.
- ["Example using the GitHub Container Registry"](https://docs-1121--patch-1.herokuapp.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-the-github-container-registry)  - Added a new section for the GitHub Container Registry.

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 1]
```diff
@@ -553,7 +553,7 @@ jobs:
   my_first_job:
     steps:
       - name: My first step
-        uses: docker://gcr.io/cloud-builders/gradle
+        uses: docker://ghcr.io/cloud-builders/gradle
```
Comment (nschonni) [inline content/actions/reference/workflow-syntax-for-github-actions.md:556]: "gcr" is for the Google Container Registry, which is probably the right one for "cloud-builders"

**Comment by janiceilene** [comment]
Thanks so much for opening a PR @varunsridharan! I'll get this triaged for review 🎉

File: content/actions/reference/workflow-syntax-for-github-actions.md (hunk 1 above)
Comment (varunsridharan) [reply content/actions/reference/workflow-syntax-for-github-actions.md:556]: Ho ok I was not aware that Google had a container registry. And in that case it should me mentioned somewhere in that place

**Comment by varunsridharan** [comment]
@janiceilene I have updated the docs once again with example on how to use from Github Container Registry & Public Registry

**Comment by varunsridharan** [comment]
Deleted Pull Request Branch By Mistake. Re-opening for review.

**Comment by varunsridharan** [comment]
@janiceilene any update on this ? will this ever be reviewed ?

**Comment by janiceilene** [comment]
Thanks for your patience @varunsridharan! Our small team is working our way through all of the amazing contributions and this PR is triaged for review.

**Comment by varunsridharan** [comment]
@github-openapi-bot still waiting 😄

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 2]
```diff
@@ -542,12 +542,26 @@ jobs:
         uses: docker://alpine:3.8
 ```
 
-##### Example using a Docker public registry action
+##### Example using a Github Container Registry
```
Comment (martin389) [inline content/actions/reference/workflow-syntax-for-github-actions.md:545]: There's some reusable text available for this product name:
```suggestion
##### Example using the {% data variables.product.prodname_github_container_registry %}
```

Comment (varunsridharan) [reply content/actions/reference/workflow-syntax-for-github-actions.md]: Cool :-) I was not aware of it.

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 3]
```diff
@@ -542,12 +542,26 @@ jobs:
         uses: docker://alpine:3.8
 ```
 
-##### Example using a Docker public registry action
+##### Example using the {% data variables.product.prodname_github_container_registry %}
 
 `docker://{host}/{image}:{tag}`
 
 A Docker image in a public registry.
```
Comment (martin389) [inline content/actions/reference/workflow-syntax-for-github-actions.md:549]:
```suggestion
A Docker image in the {% data variables.product.prodname_github_container_registry %}.
```

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 4]
```diff
@@ -542,12 +542,26 @@ jobs:
...
+    steps:
+      - name: My first step
+        uses: docker://ghcr.io/cloud-builders/gradle
+```
+
+##### Example using a Docker public registry action
+
+`docker://{host}/{image}:{tag}`
+
+A Docker image in a public registry - <sub>[Google Container Registry - GCR](https://cloud.google.com/container-registry)</sub>
```
Comment (martin389) [inline content/actions/reference/workflow-syntax-for-github-actions.md:563]:
```suggestion
A Docker image in a public registry. This example uses the Google Container Registry at `gcr.io`.
```

**Review by martin389 — COMMENTED** [review]
Thanks so much for this contribution @varunsridharan 🎊 
I've added some small suggestions while I continue checking a few things.

**Comment by martin389** [comment]
A rendered preview of these changes is available here:

- [Example using the GitHub Container Registry](https://docs-1121--patch-1.herokuapp.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-the-github-container-registry)
- [Example using a Docker public registry action](https://docs-1121--patch-1.herokuapp.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-a-docker-public-registry-action)

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 5]
```diff
@@ -542,11 +542,25 @@ jobs:
...
 
-A Docker image in a public registry.
+A Docker image in the {% data variables.product.prodname_github_container_registry %}.
+
+```yaml
+jobs:
+  my_first_job:
+    steps:
+      - name: My first step
+        uses: docker://ghcr.io/cloud-builders/gradle
```
Comment (martin389) [inline content/actions/reference/workflow-syntax-for-github-actions.md:556]: We could use the existing example from the Packages docs: https://docs.github.com/en/free-pro-team@latest/packages/guides/pushing-and-pulling-docker-images#pulling-container-images
```suggestion
        uses: docker://ghcr.io/OWNER/IMAGE_NAME
```

**Comment by martin389** [comment]
@varunsridharan - I've now updated the PR description and title to match the new changes you've added 👍

**Comment by martin389** [comment]
Checking this approach with the Packages team.

**Comment by martin389** [comment]
Following up with support folks

**Comment by martin389** [comment]
Following up with support folks ❤️

**Comment by martin389** [comment]
Support folks have confirmed they are reviewing.

**Comment by varunsridharan** [comment]
Still Waiting for Review !! ?

**Comment by martin389** [comment]
Sorry about the delay @varunsridharan! I've confirmed that this is in the queue for a review 👍

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 6]
```diff
@@ -590,11 +590,25 @@ jobs:
         uses: docker://alpine:3.8
 ```
 
-#### Example using a Docker public registry action
+##### Example using the {% data variables.product.prodname_github_container_registry %}
```
Comment (jmarlena) [inline content/actions/reference/workflow-syntax-for-github-actions.md:593]:
```suggestion
{% if currentVersion == "free-pro-team@latest" %}
##### Example using the {% data variables.product.prodname_github_container_registry %}
```

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 7]
```diff
@@ -590,11 +590,25 @@ jobs:
...
+A Docker image in the {% data variables.product.prodname_github_container_registry %}.
+
+```yaml
+jobs:
+  my_first_job:
+    steps:
+      - name: My first step
+        uses: docker://ghcr.io/OWNER/IMAGE_NAME
+```
+
```
Comment (jmarlena) [inline content/actions/reference/workflow-syntax-for-github-actions.md:606]:
```suggestion
{% endif %}
```

**Review by jmarlena — DISMISSED** [review]
@varunsridharan Thanks for your patience and sorry for the delay in reviewing this PR. 

I added some versioning around the ghcr.io mention but otherwise this should be ready to merge!

**Review by jmarlena — APPROVED** [review]
There's an outstanding product question I have about container registry support in workflows but this can merge in the meantime since it's so small scope.

**Review by Theonly797 — COMMENTED** [review]
Approved

**Pull request merged**